### DISCUSSION TITLE
setting candlePeriod

### DISCUSSION THREAD

**Discussion by CarlosASilveira** [altangent/ccxws#263] [Q&A] [answered]
I have a silly question, how to use the "candlePeriod" in the subscribeCandles module? (1m,2m,3m,5m,1h ....)
Thanks so much!

**Comment by bmancini55** [comment, marked as answer]
This is a  bit clunky at the current moment.  You need to set the candle period on the client prior to calling subscribe. The candle period then applies to all subscriptions. Unfortunately you can only set one candle period per client.  This is something that will be improved in future versions of the library.

```javascript
const binance = new ccxws.Binance();
binance.candlePeriod = CandlePeriod._1d;
binance.subscribeCandles({ id: "BTC_USDT", base: "BTC", quote: "USDT", type: "spot" });
binance.on("candle", (candle, market) => {
  // do something with market
});
```

**Reply by CarlosASilveira** [reply]
Thank you very much,

I forked and tweaked it directly in the class. Getting dynamic.
```

const period = '_1d';
const binance = new BinanceFuturesUsdtmClient({
           useAggTrades: true,
           requestSnapshot: true,
           socketBatchSize: 200,
           socketThrottleMs: 1000,
           restThrottleMs: 1000,
           CandlePeriodMod: period,
         });
```

**Reply by nileio** [reply]
@CarlosASilveira how is that any different from setting the candlePeriod prior to subscribe !? candlePeriod should actually be passed in as a parameter to `subscribeCandles` for it to be useful and not needing to create multiple clients. Now this is even worse than what it is now because you cannot re-use the client and set that property only but rather creating new clients ! am I missing something here??

**Reply by CarlosASilveira** [reply]
Nileio, the effect is the same, in the past I didn't know the period set so I made the modification in the class to make dynamic.

**Reply by nileio** [reply]
@CarlosASilveira https://github.com/nileio/ccxws/commit/ac9fe11f5df0f769b45d0b84713b6a5618336170
